python/rikai/video.py

```diff
@@ -0,0 +1,137 @@
+from abc import ABC, abstractmethod
+import cv2
+import youtube_dl
+
+from rikai.spark.types import (
+    YouTubeVideoType,
+    VideoStreamType,
+)
+from rikai.vision import Image
+
+
+class YouTubeVideo:
+
+    __UDT__ = YouTubeVideoType()
+
+    def __init__(self, vid: str):
+        self.vid = vid
+        self.uri = "https://www.youtube.com/watch?v={0}".format(self.vid)
+        self.embed_url = "http://www.youtube.com/embed/{0}".format(self.vid)
+
+    def __repr__(self) -> str:
+        return "YouTubeVideo({0})".format(self.vid)
+
+    def _repr_html_(self):
+        from IPython.lib.display import YouTubeVideo
+
+        return YouTubeVideo(self.vid)._repr_html_()
+
+    def __eq__(self, other) -> bool:
+        return isinstance(other, YouTubeVideo) and self.vid == other.vid
+
+    def get_stream(self, **kwargs) -> "VideoStream":
+        # TODO allow configurable cache dir
+        dl_path = "/tmp/{}.mp4".format(self.vid)
```

Comment:
use `TMPDIR`?

Reply:
good idea. I will use TMPDIR for this PR. I need to make this more configurable and also automatically add an option to cp to s3 as well.